File: LocationApp.py
from datetime import date, timedelta

class Objet:
    def __init__(self, nom):
        self.nom = nom
        self.disponible = True
        self.date_retour = None

    def reserver(self, jours):
        if not self.disponible:
            raise Exception(f"{self.nom} est déjà réservé jusqu'au {self.date_retour}")
        self._mettre_a_jour_reservation(jours)

    def verifier_disponibilite(self):
        if self.date_retour and date.today() >= self.date_retour:
            self._liberer()

    def _mettre_a_jour_reservation(self, jours):
        self.disponible = False
        self.date_retour = date.today() + timedelta(days=jours)
        print(f"{self.nom} réservé pour {jours} jour(s). Retour le {self.date_retour}")

    def _liberer(self):
        self.disponible = True
        self.date_retour = None

# Exemple d'utilisation
if __name__ == "__main__":
    voiture = Objet("Voiture")
    voiture.reserver(3)
    voiture.verifier_disponibilite()

File: LocationApp.cs
using System;

class Objet
{
    public string Nom { get; }
    public bool Disponible { get; private set; } = true;
    public DateTime? DateRetour { get; private set; }

    public Objet(string nom)
    {
        Nom = nom;
    }

    public void Reserver(int jours)
    {
        if (!Disponible)
        {
            throw new InvalidOperationException($"{Nom} est déjà réservé jusqu'au {DateRetour}");
        }
        MettreAJourReservation(jours);
    }

    public void VerifierDisponibilite()
    {
        if (DateRetour.HasValue && DateTime.Now >= DateRetour.Value)
        {
            Liberer();
        }
    }

    private void MettreAJourReservation(int jours)
    {
        Disponible = false;
        DateRetour = DateTime.Now.AddDays(jours);
        Console.WriteLine($"{Nom} réservé pour {jours} jour(s). Retour le {DateRetour.Value.ToShortDateString()}");
    }

    private void Liberer()
    {
        Disponible = true;
        DateRetour = null;
    }
}

class Program
{
    static void Main()
    {
        var maison = new Objet("Maison");
        maison.Reserver(4);
        maison.VerifierDisponibilite();
    }
}